packages/nextjs/hooks/scaffold-eth/useTargetNetwork.ts

```diff
@@ -1,24 +1,26 @@
 import { useEffect, useMemo } from "react";
 import { useAccount } from "wagmi";
-import { useGlobalState } from "~~/services/store/store";
-import { ChainWithAttributes, getTargetNetworks } from "~~/utils/scaffold-eth";
+import { useAbiNinjaState, useGlobalState } from "~~/services/store/store";
+import { ChainWithAttributes } from "~~/utils/scaffold-eth";
 import { NETWORKS_EXTRA_DATA } from "~~/utils/scaffold-eth";
 
-const mainNetworks = getTargetNetworks();
 /**
  * Retrieves the connected wallet's network from scaffold.config or defaults to the 0th network in the list if the wallet is not connected.
  */
 export function useTargetNetwork(): { targetNetwork: ChainWithAttributes } {
   const { chain } = useAccount();
-  const targetNetwork = useGlobalState(({ targetNetwork }) => targetNetwork);
-  const setTargetNetwork = useGlobalState(({ setTargetNetwork }) => setTargetNetwork);
+  const targetNetwork = useGlobalState(state => state.targetNetwork);
+  const setTargetNetwork = useGlobalState(state => state.setTargetNetwork);
+  const chains = useGlobalState(state => state.chains);
+  const mainChainId = useAbiNinjaState(state => state.mainChainId);
 
   useEffect(() => {
-    const newSelectedNetwork = mainNetworks.find(targetNetwork => targetNetwork.id === chain?.id);
+    const newSelectedNetwork =
+      chains.find(network => network.id === chain?.id) || chains.find(network => network.id === mainChainId);
```

Comment:
Instead of two find can't we just do?  

```ts
const newSelectedNetwork = chains.find(network => 
  network.id === chain?.id || network.id === mainChainId
);
```

Reply:
Thank you! I didn't notice that it could be done like this 🫡